commit 1eeddb72efcab49cedb457f5b6cad755cee9fe01 (refs/changes/59/2959/2)
Author: Richard Kiene <richard.kiene@joyent.com>
Date:   2017-11-17T00:12:38+00:00 (1 year, 11 months ago)
    
    joyent/manta-thoth#156 64bit node binaries are not debuggable by mdb when a 32bit node binary is in PATH

diff --git a/bin/thoth b/bin/thoth
index 2ac3141..3e3e5cc 100755
--- a/bin/thoth
+++ b/bin/thoth
@@ -1777,7 +1777,13 @@ handlers.debug = function (client, argv)
 		if (analyzer) {
 			cmd += analyzerCmd(args);
 		} else {
-			cmd += 'mdb $MANTA_INPUT_FILE';
+			/*
+			 * LIBPROC_INCORE_ELF=1 prevents us from loading
+			 * whatever node binary happens to be in $PATH. This is
+			 * necessary when debugging cores that do not match the
+			 * bitness of the node binary found in $PATH.
+			 */
+			cmd += 'LIBPROC_INCORE_ELF=1 mdb $MANTA_INPUT_FILE';
 		}
 
 		setAnalyzer(client, cmd, function (asset) {
diff --git a/package.json b/package.json
index 3fcb1fc..85dc3bf 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "manta-thoth",
 	"description": "Manta-based system for core and crash dump analysis",
-	"version": "1.0.0",
+	"version": "1.1.0",
 	"author": "Joyent (joyent.com)",
 	"dependencies": {
 		"manta": "^4.0.0",
